bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/LogEventMessage.java

```diff
@@ -0,0 +1,100 @@
+            "Cancel", "Ground fault", "Ground fault restore", "Manual test", "Closed with zones bypassed",
+            "Start of listen in", "Technician on site", "Technician left", "Control power up", "58-", "", "", "", "",
+            "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "",
+            "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "",
+            "", "", "Not used", "First to open", "Last to close", "PIN entered with bit 7 set", "Begin walk-test",
+            "End walk-test", "Re-exit", "Output trip", "Data lost" };
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+
+        // Date
+        sb.append(String.format("%02d", Integer.parseInt(day))).append('-')
+                .append(String.format("%02d", Integer.parseInt(month))).append(' ')
+                .append(String.format("%02d", Integer.parseInt(hour))).append(':')
+                .append(String.format("%02d", Integer.parseInt(minute))).append(' ');
+
+        int eventType = Integer.parseInt(type);
+        if (eventType >= 0 && eventType <= 12) { // Zone, Yes
+            sb.append(alarmList[eventType]).append(" Partition ").append(Integer.parseInt(partition) + 1)
+                    .append(" Zone ").append(Integer.parseInt(zud) + 1);
+        } else if ((eventType >= 17 && eventType <= 17) || (eventType >= 34 && eventType <= 35)
+                || (eventType >= 38 && eventType <= 39) || (eventType >= 44 && eventType <= 48)
+                || (eventType >= 50 && eventType <= 52) || (eventType >= 54 && eventType <= 57)
+                || (eventType >= 123 && eventType <= 124) || (eventType >= 127 && eventType <= 127)) { // None, No
+            sb.append(alarmList[eventType]);
+        } else if ((eventType >= 18 && eventType <= 20) || (eventType >= 22 && eventType <= 23)
+                || (eventType >= 125 && eventType <= 125)) { // None, Yes
+            sb.append(alarmList[eventType]).append(" Partition ").append(Integer.parseInt(partition) + 1);
+        } else if ((eventType >= 24 && eventType <= 33) || (eventType >= 36 && eventType <= 37)) { // Device, No
+            sb.append(alarmList[eventType]).append(" Device ").append(zud);
+        } else if ((eventType >= 40 && eventType <= 43) || (eventType >= 49 && eventType <= 49)
+                || (eventType >= 53 && eventType <= 53) || (eventType >= 120 && eventType <= 122)) { // User, Yes
+            sb.append(alarmList[eventType]).append(" Partition ").append(Integer.parseInt(partition) + 1)
+                    .append(" User ").append(Integer.parseInt(zud) + 1);
+        } else if ((eventType >= 126 && eventType <= 126)) { // User, No
+            sb.append(alarmList[eventType]).append(" User ").append(Integer.parseInt(zud) + 1);
+        }
```

Comment:
I started with a switch but it was too long and you needed to scroll down and you could not oversee it. I decided to write it like this, so that the whole logic is on one screen. I even kept the expressions the same e.g. (eventType >= 17 && eventType <= 17) so that in whole it is more readable.

Reply:
Whole class has been refactored for better readability